### ISSUE TITLE
EFI for DW1820A 00JT494

### ISSUE THREAD

**Issue by i3p9** [EchoEsprit/Hackintosh-Catalina-OpenCore-Lenovo-T450s-efi#36]
I got the exact same card as you but I can't seem to make Bluetooth/WiFi work. 

Did you just add the kexts in the DW1820 folder and did nothing else? Or did you need to add the PCI-E entries to the Devices section? 

I'm very confused with multiple answers regarding this card so if you could share your current EFI that'd be amazing. Thanks!

Edit: When used your EFI with the kexts added from dw1820a folder, the laptop boots but in a black screen and nothing else happens.

Managed to get Bluetooth to work perfectly by putting in the BT kexts (BrcmBluetoothInjector.kext, BrcmFirmwareData.kext and BrcmPatchRAM3.kext) and disabling Wireless LAN from BIOS. 

Need to know the Pci-e id of the wifi of someone with T450s, I think I'm messing up the pci-e id thing when adding it to devices. Only thing missing is now the WiFi. 

Edit: More progress, managed to boot with WiFi enabled in BIOS by adding the brcm boot args. Now it shows up in the PCI-E Slot but wifi isn't working at all. Can't turn in on or off and can't find any networks. 

<img width="964" alt="Screen Shot 2020-03-18 at 2 36 39 PM" src="https://user-images.githubusercontent.com/8825262/76941458-483f4080-6926-11ea-89c4-eff1fc3cc13d.png">
<img width="964" alt="Screen Shot 2020-03-18 at 2 36 03 PM" src="https://user-images.githubusercontent.com/8825262/76941467-4bd2c780-6926-11ea-8107-e5a3640a42d8.png">
<img width="780" alt="Screen Shot 2020-03-18 at 2 38 16 PM" src="https://user-images.githubusercontent.com/8825262/76941477-51c8a880-6926-11ea-8925-aa8bc2ac70db.png">
<img width="1392" alt="Screen Shot 2020-03-18 at 2 40 27 PM" src="https://user-images.githubusercontent.com/8825262/76941601-8ccadc00-6926-11ea-9a81-9f9795241df0.png">
And not showing up as network device in Hackintool
<img width="1392" alt="Screen Shot 2020-03-18 at 3 22 07 PM" src="https://user-images.githubusercontent.com/8825262/76945153-5a23e200-692c-11ea-997d-e0367d3b9604.png">

**Comment by androidkillers** [comment]
是否屏蔽引脚

**Comment by m4ak** [comment]
I see that you have updated OC and driver at the same time. My model is the same as Echo model, and the wireless network card is DW1820A 0VW3T3
Now there is a new driving method You only need AirportBrcmFixup.kext + boot-arg:brcmfx-driver=1 (or brcmfx-driver=2) Use AirPort.BrcmNIC to drive, you can try it. link:
https://github.com/acidanthera/AirportBrcmFixup/pull/7

**Comment by i3p9** [comment]
@cchyswy hey thank you for the help, I actually gave up on the card. But I might try again. 

I understand what you’ve told me to do but I don’t understand the ‘use AirPort.BrcmNIC to drive’ part, what do you mean by that?


And yes I have updated to OC 0.6.1 in my fork since Echo doesn’t seem to update them anymore.

**Comment by m4ak** [comment]
@i3p9 In the latest Big Sur system, the AirPortBrcm4360 driver has been removed, leaving only the airport.brcmNIC
Mentioned in https://github.com/acidanthera/AirportBrcmFixup，brcmfx-driver=0|1|2|3 enables only one kext for loading, 0 - AirPortBrcmNIC-MFG, 1 - AirPortBrcm4360, 2 - AirPortBrcmNIC, 3 - AirPortBrcm4331,So use brcmfx-driver=2
I just updated your EFI, but encountered a problem when booting:
0SCB:No suitablesignature- Securlty Vlolation
0CB: Apple Secure Boot prohibits this boot entry, enforcing

**Comment by i3p9** [comment]
@cchyswy Hey so the problem was since 0.6.1, `AppleSecureBoot` was implemented. It worked for me but for you, in config.plist, change `Misc.->Security->SecureBootModel` value to `Disabled`. That should work!

I've also updated the changed in my repo here 215d2901d1ff87f8d79cba1c7edf02003b482f1f. Thanks for testing it for me! Let me know if it works.

**Comment by m4ak** [comment]
@i3p9 Yesterday accidentally updated to Big Sur (20A5323I) version, use this EFI to boot，Misc.->Security->SecureBootModel has been changed to Disabled。
Personally, I feel that the configuration of this notebook i5-5200u 8g mem 256ssd runs Mojave relatively smoothly, but Catalina and Big Sur are not so smooth, and I feel a bit slow.

**Comment by i3p9** [comment]
@cchyswy okay this is gonna sound insane, but OC 0.6.1 made the system a little bit smoother compared to the old OC that Echo has in this repo. I noticed it almost instantly. 

And you’re right, actually. This machine is getting a bit outdated to run Catalina/Big Sur. 

Although if you already don’t have a fast ssd with dram, please upgrade! And I have 12GB ram instead of 8 which makes a difference when browsing tons of chrome tabs.

**Comment by m4ak** [comment]
@i3p9  So far, Big Sur Bate6 has been successfully guided. Today, Big Sur Bate7 has been pushed several times and is being upgraded. I don't know if there will be any problems.

Is it normal to sleep in Catalina? I didn't pay attention to this problem during the experience of Big Sur.

And I want to use this to guide Mojave, do I need to change config and kext? Now it is a dual system of Big Sur and Mojave. Mojave can be booted. It seems to be working properly. If there is no problem, I plan to reinstall the system and only install Mojave.

**Comment by i3p9** [comment]
@cchyswy yes sleep works for me in catalina. Haven’t had issues with mine. Do you have issues with your sleep?

Honestly, I got the laptop when Catalina was available and I’ve never used Mojave with this config. But I think it’s gonna work with Mojave. There isn’t really any OS version specific config in OpenCore. 

If you do wanna test Mojave, please report back! It’d be good to know if it works or not. As Mojave is a pretty important Version, being the last 32bit supported OS and all. :D
